docs/aws-go-sdk-migrations.md

```diff
@@ -0,0 +1,257 @@
+# AWS SDK For Go Migration Guide
+
+AWS has [announced](https://aws.amazon.com/blogs/developer/announcing-end-of-support-for-aws-sdk-for-go-v1-on-july-31-2025/) that the V1 of the SDK for Go will enter maintenance mode on July 31, 2024 and reach end-of-life July 31, 2025.
+While the AWS Terraform provider already [requires all net-new services](./aws-go-sdk-versions.md) to use AWS SDK V2 service clients, there remains a substantial number of existing services utilizing V1.
+
+Over time maintainers will be migrating impacted services to adopt AWS SDK for Go V2.
+For community members interested in contributoring to this effort, this guide documents the common patterns required to migrate a service.
+
+## Pre-Requisites
+
+### Re-generate Service Client
+
+When fully replacing the client, [`names/data/names_data.csv`](https://github.com/hashicorp/terraform-provider-aws/blob/main/names/data/names_data.csv) should be updated to remove the v1 indicator and add v2 (ie. delete the `1` in the `ClientSDKV1` column and add a `2` in the `ClientSDKV2` column).
+Once complete, re-generate the client.
+
+```console
+go generate ./internal/conns/...
+```
+
+### Add an `EndpointID` Constant
+
+When a service is first migrated, a `{ServiceName}EndpointID` constant must be added to [`names/names.go`](https://github.com/hashicorp/terraform-provider-aws/blob/main/names/names.go) manually.
```

Comment:
This is no longer the case: `acctest.ErrorCheck` should take the corresponding `names.<Service>ServiceID`, which is generated. It may still be needed in some cases, such as `acctest.PreCheckPartitionHasService`

Reply:
Thanks for the clarification. I'll fix the syntax in the acceptance testing ErrorCheck section below.

Given `skaff` outputs acceptance tests that use the `acctest.PreCheckPartitionHasService` PreCheck, and most existing services use it, I think this section may be worth keeping.